DuckDuckGo/HomeRowReminder.swift

```diff
@@ -54,13 +60,21 @@ class HomeRowReminder {
         return storage.shown
     }
 
-    private func hasReminderTimeElapsed() -> Bool {
-        guard let date = storage.firstAccessDate else {
-            storage.firstAccessDate = Date()
-            return false
+    private func hasReminderTimeElapsed(isDefaultBrowserSupported: Bool) -> Bool {
+        if isDefaultBrowserSupported {
+            guard let date = homeMessageStorage.dateDismissed(forHomeMessage: .defaultBrowserPrompt) else {
+                return false
+            }
+            let days = abs(date.timeIntervalSinceNow / 24 / 60 / 60)
+            return days > Constants.reminderTimeInDaysIOS14
+        } else {
+            guard let date = storage.firstAccessDate else {
+                storage.firstAccessDate = Date()
+                return false
+            }
+            let days = abs(date.timeIntervalSinceNow / 24 / 60 / 60)
+            return days > Constants.reminderTimeInDays
```

Comment:
As above, if possible.

Reply:
This one's interesting because it has a side effect (of storing the first access date if it hasn't been stored before). I can move it, but the tests use a mocked version of the storage and test that this side effect works correctly, which if I mock will undermine the tests somewhat.
 
I could maybe restructure things, but the way `HomeRowReminder` and `HomeRowReminderStorage` are structured made me think the intention was that `HomeRowReminderStorage` would always just be a thin UserDefaults wrapper, with any logic in `HomeRowReminder`

Reply:
Of course, that makes total sense.